pallets/summary/src/lib.rs

```diff
@@ -751,7 +751,7 @@ pub mod pallet {
                 safe_add_block_numbers::<BlockNumberFor<T>>(Self::current_slot(), 1u32.into())
                     .map_err(|_| Error::<T>::Overflow)?;
 
-            let new_validator_account_id = AVN::<T>::advance_primary_validator(OperationType::Avn)?;
+            let new_validator_account_id = AVN::<T>::advance_primary_validator_for_sending()?;
```

Comment:
I don't think we should be doing that here. The index is only incremented when sending I think.